### PR TITLE
:sparkles: router

### PR DESCRIPTION
**Pull request by bpetetot**
To add the "documentation screen" (#60), it's better to manage navigation through a router.
I have added the following libraries : 
- `redux-little-router`
- `hoc-little-router` (very usefull ;)

For now, I have initialized the router in "hash mode" (`routerForHash`)
I have also organized the `components`folder like that : 
```
 components/
     common/
          button/
          navbar/
          navButton/
     documentation/
     workbench/
```
And I have added the `appLayout.jsx` file managing app layout and routes.

### CODE REVIEW

File: src/gui/components/common/navbar/navbar.styles.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const navActions = css({
+  display: 'flex',
+  alignItems: 'center',
+  marginLeft: 'auto',
+  paddingRight: '1em',
+  '> button': {
+    padding: '.5em',
+    marginLeft: '1em',
+    border: '0px solid #fff',
+    borderRadius: '3px',
```
Comment (fabienjuif) [inline src/gui/components/common/navbar/navbar.styles.js]: em ?

File: src/gui/components/documentation/documentation.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+const Documentation = () => {
+  return (
+    <div>
+      Documentation
+    </div>
+  )
+}
+
+export default router('DOCUMENTATION', { absolute: true })(Documentation)
```
Comment (fabienjuif) [inline src/gui/components/documentation/documentation.jsx]: We can let it relative I think.

**Review by fabienjuif — COMMENTED** [review]
In our last projects, @guillaumecrespel and I do the following hierarchy : 
  - gui
    - components/ (no common)
    - screens/ (wich assembly components, and are connected via hoc-little-router)

**Comment by bpetetot** [comment]
👌 @fabienjuif

**Review by fabienjuif — APPROVED** [review]
👌 great

**Pull request merged**